File: PRS-bootcamp/Models/Vendor.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Web;

namespace PRS_bootcamp.Models
{
    public class Vendor
    {
        public int Id { get; set; }

        [Required]
        [StringLength(10)]
        [Index(IsUnique = true)]
        public string Code { get; set; }

        [Required]
        [StringLength(255)]
        public string Name { get; set; }

        [Required]
        [StringLength(255)]
        public string Address { get; set; }

        [Required]
        [StringLength(255)]
        public string City { get; set; }

        [Required]
        [StringLength(2)]
        public string State { get; set; }

        [Required]
        [StringLength(5)]
        public string Zip { get; set; }

        [Required]
        [StringLength(12)]
        public string Phone { get; set; }

        [Required]
        [StringLength(100)]
        public string Email { get; set; }

        public bool IsPreapproved { get; set; }

        public void Copy(Vendor vendor)
        {
            Code = vendor.Code;
            Name = vendor.Name;
            Address = vendor.Address;
            City = vendor.City;
            State = vendor.State;
            Zip = vendor.Zip;
            Phone = vendor.Phone;
            Email = vendor.Email;
            IsPreapproved = vendor.IsPreapproved;
        }
    }
}

File: src/client-ng/src/app/models/vendor.ts
export class Vendor
{
    Id: number;
    Code: string;
    Name: string;
    Address: string;
    City: string;
    State: string;
    Zip: string;
    Phone: string;
    Email: string;
    IsPreapproved: boolean;

    constructor(
        id: number,
        code: string,
        name: string,
        address: string,
        city: string,
        state: string,
        zip: string,
        phone: string,
        email: string,
        isPreapproved: boolean
    )
    {
        this.Id = id;
        this.Code = code;
        this.Name = name;
        this.Address = address;
        this.City = city;
        this.State = state;
        this.Zip = zip;
        this.Phone = phone;
        this.Email = email;
        this.IsPreapproved = isPreapproved;
    }
}
